components/compliance-service/inspec-agent/resolver/resolver.go

```diff
@@ -88,9 +87,69 @@ func (r *Resolver) ResolveJob(ctx context.Context, job *jobs.Job) ([]*types.Insp
 		logrus.Warnf("Nothing to do for job %s, empty list of nodes/selectors", job.Id)
 		return []*types.InspecJob{}, fmt.Errorf("cannot process this job, nodes and selectors are both nil")
 	}
-	r.scannerServer.UpdateJobNodeCount(job.Id, len(nodeJobs))
+	err = r.scannerServer.UpdateJobNodeCount(job.Id, len(nodeJobs))
+	if err != nil {
+		logrus.Errorf("error trying to job node count for job %s (%s): %s", job.Name, job.Id, err.Error())
+	}
 	return nodeJobs, nil
 }
+
+type nodeInfo struct {
+	UUID    string
+	CloudID string
+	// this is the aws instance id or azure vm id for traditional nodes.
+	// for azure api, it's a subscription id. for aws-api (new) it's an account id. for the old multinode path of aws
+	// api it's region. for gcp it's the project name.
+	Name              string
+	Environment       string
+	CloudAccountID    string // this is the account id on the manager object associated with the node
+	ManagerID         string // the id of the manager associated with the node
+	SSM               bool   // true means we should treat this as a ssm job (remote scan job)
+	MachineIdentifier string // special identifier used when querying azure api, the resource group name
+	Tags              []*common.Kv
+}
+
+func assembleJob(job *jobs.Job, node nodeInfo, secrets []*inspec.Secrets, tc inspec.TargetBaseConfig) *types.InspecJob {
+	if len(node.UUID) == 0 {
+		id, err := uuid.NewV4()
+		if err != nil {
+			logrus.Warnf("unable to generate uuid for node")
+		}
```

Comment:
Is it ok to continue processing if this fails? Can this error get bubbled up?

Reply:
oh good point. if we continue on from here we’ll eventually get a scan job report that gets rejected. i’ll push an update for that!